<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Introduction to RNAseq</title>
  <meta name="description" content="Introduction to RNAseq">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Introduction to RNAseq" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to RNAseq" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="gene-annotation.html">
<link rel="next" href="gff-1-2-and-3.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction to RNAseq</a></li>
<li class="chapter" data-level="1" data-path="genomics.html"><a href="genomics.html"><i class="fa fa-check"></i><b>1</b> Genomics</a><ul>
<li class="chapter" data-level="1.1" data-path="genomics.html"><a href="genomics.html#sequence"><i class="fa fa-check"></i><b>1.1</b> Sequence</a></li>
<li class="chapter" data-level="1.2" data-path="genomics.html"><a href="genomics.html#read-sequence"><i class="fa fa-check"></i><b>1.2</b> Read sequence</a></li>
<li class="chapter" data-level="1.3" data-path="genomics.html"><a href="genomics.html#contig-sequence"><i class="fa fa-check"></i><b>1.3</b> Contig sequence</a></li>
<li class="chapter" data-level="1.4" data-path="genomics.html"><a href="genomics.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="reference-sequence.html"><a href="reference-sequence.html"><i class="fa fa-check"></i><b>2</b> Reference sequence</a><ul>
<li class="chapter" data-level="2.1" data-path="reference-sequence.html"><a href="reference-sequence.html#exercise"><i class="fa fa-check"></i><b>2.1</b> Exercise</a></li>
<li class="chapter" data-level="2.2" data-path="reference-sequence.html"><a href="reference-sequence.html#rule-of-thumb"><i class="fa fa-check"></i><b>2.2</b> Rule of thumb</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>3</b> Gene annotation</a></li>
<li class="chapter" data-level="4" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html"><i class="fa fa-check"></i><b>4</b> Bioinformatics file formats</a><ul>
<li class="chapter" data-level="4.1" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#what-is-a-file-format"><i class="fa fa-check"></i><b>4.2</b> What is a file format?</a></li>
<li class="chapter" data-level="4.3" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#annotation-files"><i class="fa fa-check"></i><b>4.3</b> Annotation files</a></li>
<li class="chapter" data-level="4.4" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#plain-text-files"><i class="fa fa-check"></i><b>4.4</b> Plain text files</a></li>
<li class="chapter" data-level="4.5" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#glossary"><i class="fa fa-check"></i><b>4.5</b> Glossary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="gff-1-2-and-3.html"><a href="gff-1-2-and-3.html"><i class="fa fa-check"></i><b>5</b> GFF 1, 2 and 3</a><ul>
<li class="chapter" data-level="5.1" data-path="gff-1-2-and-3.html"><a href="gff-1-2-and-3.html#general-feature-format"><i class="fa fa-check"></i><b>5.1</b> General Feature Format</a></li>
<li class="chapter" data-level="5.2" data-path="gff-1-2-and-3.html"><a href="gff-1-2-and-3.html#general-feature-format-version-1-gff1"><i class="fa fa-check"></i><b>5.2</b> General Feature Format Version 1 (GFF1)</a></li>
<li class="chapter" data-level="5.3" data-path="gff-1-2-and-3.html"><a href="gff-1-2-and-3.html#general-feature-format-version-2-gff2"><i class="fa fa-check"></i><b>5.3</b> General Feature Format Version 2 (GFF2)</a></li>
<li class="chapter" data-level="5.4" data-path="gff-1-2-and-3.html"><a href="gff-1-2-and-3.html#general-feature-format-version-3-gff3"><i class="fa fa-check"></i><b>5.4</b> General Feature Format Version 3 (GFF3)</a></li>
<li class="chapter" data-level="5.5" data-path="gff-1-2-and-3.html"><a href="gff-1-2-and-3.html#gene-transfer-format-gtf"><i class="fa fa-check"></i><b>5.5</b> Gene Transfer Format (GTF)</a></li>
<li class="chapter" data-level="5.6" data-path="gff-1-2-and-3.html"><a href="gff-1-2-and-3.html#how-to-parse-gffgtf-file.."><i class="fa fa-check"></i><b>5.6</b> How to parse GFF/GTF file..?</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html"><i class="fa fa-check"></i><b>6</b> All about RNAseq</a><ul>
<li class="chapter" data-level="6.1" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#general"><i class="fa fa-check"></i><b>6.1</b> General</a></li>
<li class="chapter" data-level="6.2" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#technology"><i class="fa fa-check"></i><b>6.2</b> Technology</a></li>
<li class="chapter" data-level="6.3" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#differential-expression-de"><i class="fa fa-check"></i><b>6.3</b> Differential expression (DE)</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="lesson-plan.html"><a href="lesson-plan.html"><i class="fa fa-check"></i><b>7</b> Lesson plan</a></li>
<li class="chapter" data-level="8" data-path="meeting-notes.html"><a href="meeting-notes.html"><i class="fa fa-check"></i><b>8</b> Meeting notes</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to RNAseq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bioinformatics-file-formats" class="section level1">
<h1><span class="header-section-number">4</span> Bioinformatics file formats</h1>
<p>Before diving into specific file formats. I would like to discuss what could file actually mean and hold in general, after all bioinformatics files aren’t that different form any other files. In fact fair number of bioinformatics files are just a variant of <a href="#tab-separated-variables">TSV file format</a>, discussed shortly. Bioinformatics world largely rotates around Unix/Linux ecosystem, and remember that macOS is Unix derived operating system too. That means that a lot of the tooling and compute happens on linux computers often referred to as servers. In unix world file name is simply a string of letters (text) that points to a location of that file content that is stored elsewhere. Therefore the only thing that matters for file names is they are unique. Often people get caught out by file’s extensions, they do not matter. File extentions are simply an indicator for a user what to expect when they are going to open that file. To more accurate other programs often look at file extensions too trying to guess if this is the file format they can handle. In general it is a very good idea to have correct file extension. I just want you to know that just because you’ve renamed a file from <code>ref.fasta</code> to <code>ref.fa</code> nothing had happened to it content. Bioinformatics is an evolving field and new technologies will bring about new file formats.</p>
<div id="introduction" class="section level2">
<h2><span class="header-section-number">4.1</span> Introduction</h2>
<p>Broadly speaking any computer file is either a plain text or binary format. Under the hood everything is binary of course, but a plain text file can be looked at and modified using standard unix/linux tool sets as well simply opened with any text editor for more custom modification. If the file isn’t plain text you have to have specific tool to work with that file type. For example pdf files aren’t plain text and this is why you need pdf viewer (e.g adobe reader) to work with pdfs. Similarly in bioinformatics BAM files aren’t plain text and do require <code>samtools</code> to work with it. As I’ve hinted earlier large number of bioinformatics files are plain text, however a lot of the times plain text files are ziped with <code>gzip</code> tool adding another suffix to the file name and thereby changing the file into binary for that now will require <code>gzip</code> tool to work with. We zip file to reduce they size. Zipping works particular well on files that have repetitive patterns e.g repeats of ATCG sequence.</p>
<p>Note the difference in size is essentially three times !</p>
<pre><code>-rw-r--r-- 1 kirill kirill 3.0G Apr 10 10:11 Homo_sapiens.GRCh38.dna_sm.removed_contigs.fa
-rw-r--r-- 1 kirill kirill 895M Apr 10 10:10 Homo_sapiens.GRCh38.dna_sm.removed_contigs.fa.gz</code></pre>
<p>It is a natural behaviour wanting to unzip file before looking inside. It is okay to do so, but be aware that this isn’t the best practice, and you don’t want to be storing your files in plain text for a long time, this will eat up your disk space very quickly.</p>
<pre><code>example of large BAM file and plain text SAM</code></pre>
<p>Remember that best practice is always keep your files zipped</p>
<p>Let’s talk next about what makes different file formats different</p>
</div>
<div id="what-is-a-file-format" class="section level2">
<h2><span class="header-section-number">4.2</span> What is a file format?</h2>
<p>The file format at it simplest defines structure with in the file. By far the most common structure of the plain text file is tabular. The spreadsheet that everyone is so familiar with is simply a plain text file with rows and columns, where each cell separated by a “special” character called <code>tabular</code> or <code>tab</code> for short, which makes tab separated variables file or tsv for short with a sister file of comma separated variables or csv for short where instead of <code>tab</code> character comma is used.</p>
<p><strong>Comma Separated Variables (csv)</strong></p>
<pre><code>Gene.ID Chrom   Gene.Name   Biotype KO3_S3  KO4_S4  WT1_S1  WT2_S2
ENSMUSG00000051951  1   Xkr4    protein_coding  0   0   0   0
ENSMUSG00000025900  1   Rp1 protein_coding  0   0   0   0
ENSMUSG00000109048  1   Rp1 protein_coding  0   0   0   0
ENSMUSG00000025902  1   Sox17   protein_coding  0   0   0   0
ENSMUSG00000033845  1   Mrpl15  protein_coding  934 1317    888 939
ENSMUSG00000025903  1   Lypla1  protein_coding  705 848 647 747
ENSMUSG00000104217  1   Gm37988 protein_coding  0   0   0   0
ENSMUSG00000033813  1   Tcea1   protein_coding  354 436 368 442
ENSMUSG00000002459  1   Rgs20   protein_coding  0   0   0   0</code></pre>
<p><strong>Tab Separated Variables (tsv)</strong></p>
<pre><code>Gene.ID,Chrom,Gene.Name,Biotype,KO3_S3,KO4_S4,WT1_S1,WT2_S2
ENSMUSG00000051951,1,Xkr4,protein_coding,0,0,0,0
ENSMUSG00000025900,1,Rp1,protein_coding,0,0,0,0
ENSMUSG00000109048,1,Rp1,protein_coding,0,0,0,0
ENSMUSG00000025902,1,Sox17,protein_coding,0,0,0,0
ENSMUSG00000033845,1,Mrpl15,protein_coding,934,1317,888,939
ENSMUSG00000025903,1,Lypla1,protein_coding,705,848,647,747
ENSMUSG00000104217,1,Gm37988,protein_coding,0,0,0,0
ENSMUSG00000033813,1,Tcea1,protein_coding,354,436,368,442
ENSMUSG00000002459,1,Rgs20,protein_coding,0,0,0,0</code></pre>
<p>Common bioinformatics files such gff/gtf, bed and vcf are all variant of tsv in some respect except with defined column names and hence value types that can fit in those columns. I haven’t explained what each of those files used for. Let’s focus on gff/gtf format only and leave the other two for later discussion.</p>
</div>
<div id="annotation-files" class="section level2">
<h2><span class="header-section-number">4.3</span> Annotation files</h2>
<p>The essence of annotation file, guessable from the name, is to provide information of coordinates of features in the genome. A feature could mean several things including a gene, transcript, exon, 3’UTR or any other region of the gnome that is “worth” documenting. Basically this is your google map without pictures. If you think about what information would you need in order to identify a gene for example well known tumor suppressor gene <code>TP53</code>.</p>
<ul>
<li>bear minimum we need to know which chromosome it is one</li>
<li>where about it is on chromosome 17</li>
<li>remember that humans are diploid - which strand of chromosome 17 it is on</li>
</ul>
<p>In a simply tabular format this would look like this</p>
<pre><code>tp53    chr17   7661779 7687550 +</code></pre>
<p>Turns out that this is actually is an annotation file format called <a href="http://bioinf.wehi.edu.au/featureCounts/">Simplified Annotation Format (SAF)</a>. It isn’t that widely used, because it has some limitations, which is outside of the course to discuss and there are at least three other annotation formats that exist, which we not going to discuss any further. Side tracking to our earlier discussion about file extensions, <code>saf</code> is a file format given that it has defined structure, but what should the extension be? Anything you like <code>.txt</code> because it is a text file, <code>.tsv</code> well it is tab separated. I usually do <code>.saf</code>, because that tells me straight away exactly what the content of the file is. This is what file extension supposed to do.</p>
<p>The file format that we are going to focus on is General Feature Format (gff)</p>
<ul>
<li>Plain text file format</li>
<li>The format was proposed as a means to transfer feature information</li>
<li>We do not intend GFF format to be used for complete data management of the analysis and annotation of genomic sequence</li>
<li>Originally wanted file format that will be easily parsable by Unix tools such <code>grep</code>, <code>sort</code> and <code>awk</code></li>
</ul>
<p>For the purpose of understanding RNAseq experiment the bear minimum you’ll need to understand are these three file types:</p>
<ul>
<li>FASTQ</li>
<li>FASTA</li>
<li>GFF/GTF</li>
</ul>
<div id="fastq" class="section level4">
<h4><span class="header-section-number">4.3.0.1</span> FASTQ</h4>
<p>This is a very important file(s) you don’t mess directly with FASTQ files. Best not to every unzip those either. If think that you need to do some thing fastq files in terms of editing or some other out of ordinary manipulations. Make a copy and work on the copy. If you modify, corrupt or loose your FASTQ files, days, month or year of wet-lab work go down a toilet, let a lone large sum of money spend on RNA extraction, library prepartion and sequencing.</p>
</div>
<div id="fasta-1" class="section level4">
<h4><span class="header-section-number">4.3.0.2</span> FASTA</h4>
<p>There are a few different fasta files, but all of them will hold some sort of sequence. Sometime you’ll see <code>.fna</code> to inidicate nuclear sequence or <code>.faa</code> to indicate amino acid sequence, but more generally it’ll be <code>.fa</code> or a longer version <code>.fasta</code>. Like any biological plain text file. This most certainly will be zipped so expect to see <code>.fa.gz</code> or <code>.fasta.gz</code>.</p>
<p>For the purpose of RNAseq analysis know that FASTA file holds your reference sequence.</p>
</div>
<div id="gtfgff" class="section level4">
<h4><span class="header-section-number">4.3.0.3</span> GTF/GFF</h4>
<p>These are annotation files</p>
<p>There are many different biology / bioinformatics related file formats. You will naturally come across different files formats. It doesn’t matter so much what they are called, i.e what they file extention is, rather what they hold.</p>
<p>Essentially all bioinformaticie files are either plain text (a.k.a ASCII) or binnary. Plain text files are “easy” to interface with. Unix/Linux command line provide many great tools that can work with plain text files. To list a few populart and powerful command line tools:</p>
<ul>
<li><code>grep</code> search for a pattern</li>
<li><code>sed</code> search and replace</li>
<li><code>awk</code> search and replace and more</li>
<li><code>cat</code> look inside</li>
<li><code>cut</code> what the knife would do to bread</li>
</ul>
<p>It is out of the scope of this course to go through any of the tools or command line in general, but if you are going to do any kind of bioinformatics you’ll eventually going to come across these tools.</p>
<p>The other file type is binnary, this will requires designated tool for interfacing with the file for example SAM/BAM files have <code>samtools</code> that aid interaction with them.</p>
</div>
</div>
<div id="plain-text-files" class="section level2">
<h2><span class="header-section-number">4.4</span> Plain text files</h2>
<p>Given that we know the structure of the file TSV or CSV we can use standard command line tools to get relevant/interesting bits e.g</p>
<pre><code>~$ cut -f1,3,5-6 t.tsv

Gene.ID Gene.Name   KO3_S3  KO4_S4
ENSMUSG00000051951  Xkr4    0   0
ENSMUSG00000025900  Rp1 0   0
ENSMUSG00000109048  Rp1 0   0
ENSMUSG00000025902  Sox17   0   0
ENSMUSG00000033845  Mrpl15  934 1317
ENSMUSG00000025903  Lypla1  705 848
ENSMUSG00000104217  Gm37988 0   0
ENSMUSG00000033813  Tcea1   354 436
ENSMUSG00000002459  Rgs20   0   0</code></pre>
</div>
<div id="glossary" class="section level2">
<h2><span class="header-section-number">4.5</span> Glossary</h2>
<ul>
<li>interface: extract or insert relevant information</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gene-annotation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gff-1-2-and-3.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["rnaseq-intro.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
