<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Introduction to RNAseq</title>
  <meta name="description" content="Introduction to RNAseq">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Introduction to RNAseq" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to RNAseq" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="gene-annotation.html">
<link rel="next" href="all-about-rnaseq.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction to RNAseq</a></li>
<li class="chapter" data-level="1" data-path="genomics.html"><a href="genomics.html"><i class="fa fa-check"></i><b>1</b> Genomics</a><ul>
<li class="chapter" data-level="1.1" data-path="genomics.html"><a href="genomics.html#sequence"><i class="fa fa-check"></i><b>1.1</b> Sequence</a></li>
<li class="chapter" data-level="1.2" data-path="genomics.html"><a href="genomics.html#read-sequence"><i class="fa fa-check"></i><b>1.2</b> Read sequence</a></li>
<li class="chapter" data-level="1.3" data-path="genomics.html"><a href="genomics.html#contig-sequence"><i class="fa fa-check"></i><b>1.3</b> Contig sequence</a></li>
<li class="chapter" data-level="1.4" data-path="genomics.html"><a href="genomics.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="reference-sequence.html"><a href="reference-sequence.html"><i class="fa fa-check"></i><b>2</b> Reference sequence</a><ul>
<li class="chapter" data-level="2.1" data-path="reference-sequence.html"><a href="reference-sequence.html#exercise"><i class="fa fa-check"></i><b>2.1</b> Exercise</a></li>
<li class="chapter" data-level="2.2" data-path="reference-sequence.html"><a href="reference-sequence.html#rule-of-thumb"><i class="fa fa-check"></i><b>2.2</b> Rule of thumb</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>3</b> Gene annotation</a></li>
<li class="chapter" data-level="4" data-path="file-formats.html"><a href="file-formats.html"><i class="fa fa-check"></i><b>4</b> File formats</a><ul>
<li class="chapter" data-level="4.1" data-path="file-formats.html"><a href="file-formats.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="file-formats.html"><a href="file-formats.html#plain-text-files"><i class="fa fa-check"></i><b>4.2</b> Plain text files</a></li>
<li class="chapter" data-level="4.3" data-path="file-formats.html"><a href="file-formats.html#glossary"><i class="fa fa-check"></i><b>4.3</b> Glossary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html"><i class="fa fa-check"></i><b>5</b> All about RNAseq</a><ul>
<li class="chapter" data-level="5.1" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#general"><i class="fa fa-check"></i><b>5.1</b> General</a></li>
<li class="chapter" data-level="5.2" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#technology"><i class="fa fa-check"></i><b>5.2</b> Technology</a></li>
<li class="chapter" data-level="5.3" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#differential-expression-de"><i class="fa fa-check"></i><b>5.3</b> Differential expression (DE)</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="lesson-plan.html"><a href="lesson-plan.html"><i class="fa fa-check"></i><b>6</b> Lesson plan</a></li>
<li class="chapter" data-level="7" data-path="meeting-notes.html"><a href="meeting-notes.html"><i class="fa fa-check"></i><b>7</b> Meeting notes</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to RNAseq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="file-formats" class="section level1">
<h1><span class="header-section-number">4</span> File formats</h1>
<div id="introduction" class="section level2">
<h2><span class="header-section-number">4.1</span> Introduction</h2>
<p>Bioinformatics file types doen’t different form any other files type. Broadly any computer file is divided into either a plain text or binary files. The type doesn’t matter; plain text files are “easy” to interact by it’s nature it is plain text, whereas binary files requrie specialised tooling for interfacing with the file. Note that it is very common to zip plain text files, which simply means reducing the size of the file, which is better for storage purposes. Once zipped file is no longer plain text - it is binary now. It is a natural behaviour to unzip those file before looking inside. It is okay to do so, but be aware that this isn’t the best practice, and you don’t want to be storing your files in plain text, this will eat up your disk space very quickly.</p>
<p>What is a file format?</p>
<p>It is defined structure with in the file. By far the most common structure of the plain text file is tabular, just like your conventional spreadsheet, where each column is separate by either a comma or a tab.</p>
<div id="comma-separated-variables-csv" class="section level3">
<h3><span class="header-section-number">4.1.1</span> Comma Separated Variables (CSV)</h3>
<pre><code>Gene.ID Chrom   Gene.Name   Biotype KO3_S3  KO4_S4  WT1_S1  WT2_S2
ENSMUSG00000051951  1   Xkr4    protein_coding  0   0   0   0
ENSMUSG00000025900  1   Rp1 protein_coding  0   0   0   0
ENSMUSG00000109048  1   Rp1 protein_coding  0   0   0   0
ENSMUSG00000025902  1   Sox17   protein_coding  0   0   0   0
ENSMUSG00000033845  1   Mrpl15  protein_coding  934 1317    888 939
ENSMUSG00000025903  1   Lypla1  protein_coding  705 848 647 747
ENSMUSG00000104217  1   Gm37988 protein_coding  0   0   0   0
ENSMUSG00000033813  1   Tcea1   protein_coding  354 436 368 442
ENSMUSG00000002459  1   Rgs20   protein_coding  0   0   0   0</code></pre>
</div>
<div id="tab-separated-variables-tsv" class="section level3">
<h3><span class="header-section-number">4.1.2</span> Tab Separated Variables (TSV)</h3>
<pre><code>Gene.ID,Chrom,Gene.Name,Biotype,KO3_S3,KO4_S4,WT1_S1,WT2_S2
ENSMUSG00000051951,1,Xkr4,protein_coding,0,0,0,0
ENSMUSG00000025900,1,Rp1,protein_coding,0,0,0,0
ENSMUSG00000109048,1,Rp1,protein_coding,0,0,0,0
ENSMUSG00000025902,1,Sox17,protein_coding,0,0,0,0
ENSMUSG00000033845,1,Mrpl15,protein_coding,934,1317,888,939
ENSMUSG00000025903,1,Lypla1,protein_coding,705,848,647,747
ENSMUSG00000104217,1,Gm37988,protein_coding,0,0,0,0
ENSMUSG00000033813,1,Tcea1,protein_coding,354,436,368,442
ENSMUSG00000002459,1,Rgs20,protein_coding,0,0,0,0</code></pre>
<p>For the purpose of understanding RNAseq experiment the bear minimum you’ll need to understand are these three file types:</p>
<ul>
<li>FASTQ</li>
<li>FASTA</li>
<li>GFF/GTF</li>
</ul>
<div id="fastq" class="section level4">
<h4><span class="header-section-number">4.1.2.1</span> FASTQ</h4>
<p>This is a very important file(s) you don’t mess directly with FASTQ files. Best not to every unzip those either. If think that you need to do some thing fastq files in terms of editing or some other out of ordinary manipulations. Make a copy and work on the copy. If you modify, corrupt or loose your FASTQ files, days, month or year of wet-lab work go down a toilet, let a lone large sum of money spend on RNA extraction, library prepartion and sequencing.</p>
</div>
<div id="fasta-1" class="section level4">
<h4><span class="header-section-number">4.1.2.2</span> FASTA</h4>
<p>There are a few different fasta files, but all of them will hold some sort of sequence. Sometime you’ll see <code>.fna</code> to inidicate nuclear sequence or <code>.faa</code> to indicate amino acid sequence, but more generally it’ll be <code>.fa</code> or a longer version <code>.fasta</code>. Like any biological plain text file. This most certainly will be zipped so expect to see <code>.fa.gz</code> or <code>.fasta.gz</code>.</p>
<p>For the purpose of RNAseq analysis know that FASTA file holds your reference sequence.</p>
</div>
<div id="gtfgff" class="section level4">
<h4><span class="header-section-number">4.1.2.3</span> GTF/GFF</h4>
<p>These are annotation files</p>
<p>There are many different biology / bioinformatics related file formats. You will naturally come across different files formats. It doesn’t matter so much what they are called, i.e what they file extention is, rather what they hold.</p>
<p>Essentially all bioinformaticie files are either plain text (a.k.a ASCII) or binnary. Plain text files are “easy” to interface with. Unix/Linux command line provide many great tools that can work with plain text files. To list a few populart and powerful command line tools:</p>
<ul>
<li><code>grep</code> search for a pattern</li>
<li><code>sed</code> search and replace</li>
<li><code>awk</code> search and replace and more</li>
<li><code>cat</code> look inside</li>
<li><code>cut</code> what the knife would do to bread</li>
</ul>
<p>It is out of the scope of this course to go through any of the tools or command line in general, but if you going to do any kind of bioifnormatics you’ll eventually going to come across these tools.</p>
<p>The other file type is binnary, this will requires designated tool for interfacing with the file for example SAM/BAM files have <code>samtools</code> that aid interaction with them.</p>
</div>
</div>
</div>
<div id="plain-text-files" class="section level2">
<h2><span class="header-section-number">4.2</span> Plain text files</h2>
<p>Given that we know the structure of the file TSV or CSV we can use standard command line tools to get relevant/interesting bits e.g</p>
<pre><code>~$ cut -f1,3,5-6 t.tsv

Gene.ID Gene.Name   KO3_S3  KO4_S4
ENSMUSG00000051951  Xkr4    0   0
ENSMUSG00000025900  Rp1 0   0
ENSMUSG00000109048  Rp1 0   0
ENSMUSG00000025902  Sox17   0   0
ENSMUSG00000033845  Mrpl15  934 1317
ENSMUSG00000025903  Lypla1  705 848
ENSMUSG00000104217  Gm37988 0   0
ENSMUSG00000033813  Tcea1   354 436
ENSMUSG00000002459  Rgs20   0   0</code></pre>
</div>
<div id="glossary" class="section level2">
<h2><span class="header-section-number">4.3</span> Glossary</h2>
<ul>
<li>interface: extract or insert relevant information</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gene-annotation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="all-about-rnaseq.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["rnaseq-intro.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
