<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Introduction to RNAseq</title>
  <meta name="description" content="Introduction to RNAseq">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Introduction to RNAseq" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to RNAseq" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="bioinformatics-file-formats.html">
<link rel="next" href="gene-annotation.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction to RNAseq</a></li>
<li class="chapter" data-level="1" data-path="genomics.html"><a href="genomics.html"><i class="fa fa-check"></i><b>1</b> Genomics</a><ul>
<li class="chapter" data-level="1.1" data-path="genomics.html"><a href="genomics.html#sequence"><i class="fa fa-check"></i><b>1.1</b> Sequence</a></li>
<li class="chapter" data-level="1.2" data-path="genomics.html"><a href="genomics.html#read-sequence"><i class="fa fa-check"></i><b>1.2</b> Read sequence</a></li>
<li class="chapter" data-level="1.3" data-path="genomics.html"><a href="genomics.html#contig-sequence"><i class="fa fa-check"></i><b>1.3</b> Contig sequence</a></li>
<li class="chapter" data-level="1.4" data-path="genomics.html"><a href="genomics.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html"><i class="fa fa-check"></i><b>2</b> Bioinformatics file formats</a><ul>
<li class="chapter" data-level="2.1" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#what-is-a-file-format"><i class="fa fa-check"></i><b>2.2</b> What is a file format?</a></li>
<li class="chapter" data-level="2.3" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#annotation-files"><i class="fa fa-check"></i><b>2.3</b> Annotation files</a></li>
<li class="chapter" data-level="2.4" data-path="bioinformatics-file-formats.html"><a href="bioinformatics-file-formats.html#glossary"><i class="fa fa-check"></i><b>2.4</b> Glossary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="reference-sequence.html"><a href="reference-sequence.html"><i class="fa fa-check"></i><b>3</b> Reference sequence</a><ul>
<li class="chapter" data-level="3.1" data-path="reference-sequence.html"><a href="reference-sequence.html#fasta"><i class="fa fa-check"></i><b>3.1</b> FASTA</a></li>
<li class="chapter" data-level="3.2" data-path="reference-sequence.html"><a href="reference-sequence.html#exercise"><i class="fa fa-check"></i><b>3.2</b> Exercise</a></li>
<li class="chapter" data-level="3.3" data-path="reference-sequence.html"><a href="reference-sequence.html#rule-of-thumb"><i class="fa fa-check"></i><b>3.3</b> Rule of thumb</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="gene-annotation.html"><a href="gene-annotation.html"><i class="fa fa-check"></i><b>4</b> Gene annotation</a></li>
<li class="chapter" data-level="5" data-path="raw-data.html"><a href="raw-data.html"><i class="fa fa-check"></i><b>5</b> Raw data</a><ul>
<li class="chapter" data-level="5.1" data-path="raw-data.html"><a href="raw-data.html#introduction-1"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="raw-data.html"><a href="raw-data.html#fastq"><i class="fa fa-check"></i><b>5.2</b> FASTQ</a></li>
<li class="chapter" data-level="5.3" data-path="raw-data.html"><a href="raw-data.html#unix-tooling"><i class="fa fa-check"></i><b>5.3</b> Unix tooling</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rnaseq-bioinformatics-workflow.html"><a href="rnaseq-bioinformatics-workflow.html"><i class="fa fa-check"></i><b>6</b> RNAseq bioinformatics workflow</a><ul>
<li class="chapter" data-level="6.1" data-path="rnaseq-bioinformatics-workflow.html"><a href="rnaseq-bioinformatics-workflow.html#adapter-trimming"><i class="fa fa-check"></i><b>6.1</b> Adapter trimming</a></li>
<li class="chapter" data-level="6.2" data-path="rnaseq-bioinformatics-workflow.html"><a href="rnaseq-bioinformatics-workflow.html#alignment"><i class="fa fa-check"></i><b>6.2</b> Alignment</a></li>
<li class="chapter" data-level="6.3" data-path="rnaseq-bioinformatics-workflow.html"><a href="rnaseq-bioinformatics-workflow.html#read-counting"><i class="fa fa-check"></i><b>6.3</b> Read counting</a></li>
<li class="chapter" data-level="6.4" data-path="rnaseq-bioinformatics-workflow.html"><a href="rnaseq-bioinformatics-workflow.html#rnaseq-checklist"><i class="fa fa-check"></i><b>6.4</b> RNAseq checklist</a></li>
<li class="chapter" data-level="6.5" data-path="rnaseq-bioinformatics-workflow.html"><a href="rnaseq-bioinformatics-workflow.html#summary-1"><i class="fa fa-check"></i><b>6.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="qc.html"><a href="qc.html"><i class="fa fa-check"></i><b>7</b> QC</a><ul>
<li class="chapter" data-level="7.1" data-path="qc.html"><a href="qc.html#number-of-mapped-reads"><i class="fa fa-check"></i><b>7.1</b> Number of mapped reads</a></li>
<li class="chapter" data-level="7.2" data-path="qc.html"><a href="qc.html#star-aligner"><i class="fa fa-check"></i><b>7.2</b> STAR aligner</a></li>
<li class="chapter" data-level="7.3" data-path="qc.html"><a href="qc.html#number-of-assigned-reads"><i class="fa fa-check"></i><b>7.3</b> Number of assigned reads</a></li>
<li class="chapter" data-level="7.4" data-path="qc.html"><a href="qc.html#fastqc"><i class="fa fa-check"></i><b>7.4</b> FastQC</a></li>
<li class="chapter" data-level="7.5" data-path="qc.html"><a href="qc.html#intragenic-vs-intergenic-region"><i class="fa fa-check"></i><b>7.5</b> Intragenic vs Intergenic region</a></li>
<li class="chapter" data-level="7.6" data-path="qc.html"><a href="qc.html#reads-mapping-bias"><i class="fa fa-check"></i><b>7.6</b> Reads mapping bias</a></li>
<li class="chapter" data-level="7.7" data-path="qc.html"><a href="qc.html#insert-size-distribution"><i class="fa fa-check"></i><b>7.7</b> Insert size distribution</a></li>
<li class="chapter" data-level="7.8" data-path="qc.html"><a href="qc.html#where-to-look-for-more-help"><i class="fa fa-check"></i><b>7.8</b> Where to look for more help</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html"><i class="fa fa-check"></i><b>8</b> All about RNAseq</a><ul>
<li class="chapter" data-level="8.1" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#general"><i class="fa fa-check"></i><b>8.1</b> General</a></li>
<li class="chapter" data-level="8.2" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#technology"><i class="fa fa-check"></i><b>8.2</b> Technology</a></li>
<li class="chapter" data-level="8.3" data-path="all-about-rnaseq.html"><a href="all-about-rnaseq.html#differential-expression-de"><i class="fa fa-check"></i><b>8.3</b> Differential expression (DE)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="lesson-plan.html"><a href="lesson-plan.html"><i class="fa fa-check"></i><b>9</b> Lesson plan</a></li>
<li class="chapter" data-level="10" data-path="meeting-notes.html"><a href="meeting-notes.html"><i class="fa fa-check"></i><b>10</b> Meeting notes</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to RNAseq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="reference-sequence" class="section level1">
<h1><span class="header-section-number">3</span> Reference sequence</h1>
<p>In this course will be focusing on DNA/RNA molecules and hence our sequences will be nucleotides sequences. In the case of <em>Mus_musculus</em> (mouse) genome, we have 19 autosomal chromosomes and two sex chromosomes X and Y plus a mitochondrial chromosome (genome). This makes up 22 separate molecules and therefore 22 separate sequences. Typically that sequence is stored in <strong>FASTA</strong> file format</p>
<ul>
<li>FASTQ for raw short reads</li>
<li>FAST5 for raw long reads</li>
<li>FASTA for any other kind of sequence</li>
</ul>
<p>The other possible bioinformatics files that we weren’t focus on are, genbank and gff. Both of those files can hold annotation and sequence information in one file. It often works fine with small genomes, but large eukaryotic genomes are better split into two files reference sequence and annotation information.</p>
<div id="fasta" class="section level2">
<h2><span class="header-section-number">3.1</span> FASTA</h2>
<p>There are a few different fasta files, but all of them will hold some sort of sequence. Sometime you’ll see <code>.fna</code> to inidicate nuclear sequence or <code>.faa</code> to indicate amino acid sequence, but more generally it’ll be <code>.fa</code> or a longer version <code>.fasta</code>. Like any biological plain text file. This most certainly will be zipped so expect to see <code>.fa.gz</code> or <code>.fasta.gz</code>.</p>
<p>For the purpose of RNAseq analysis know that FASTA file holds your reference sequence.</p>
<p>FASTA file typically can hold any type of sequence data. We are not going to talk about amino acid sequence in this course and so for your purpose we can think of FASTA as just holding nucleotides sequence. That sequence can be anything, primer or adapter sequence, individual transcripts (genes) sequence OR entire chromosome sequence. In the case when FASTA holds entire genome, that is all chromosomes for a given species, we tend to refer to that FASTA file as a reference genome file. If I’m asking biologist of a <strong>reference genome</strong>, I typically imply that I need a FASTA file with genomic sequence.</p>
<p>Be aware that there isn’t such thing as “the genome” for a particular species. This is because any genome is a</p>
<p>There are a few different fasta files, but all of them will hold some sort of sequence. Sometime you’ll see <code>.fna</code> to inidicate nuclear sequence or <code>.faa</code> to indicate amino acid sequence, but more generally it’ll be <code>.fa</code> or a longer version <code>.fasta</code>. Like any biological plain text file. This most certainly will be zipped so expect to see <code>.fa.gz</code> or <code>.fasta.gz</code>.</p>
<p>Note that our purpose we can think of FASTA file holding our reference genome sequence of intereste e.g mouse genome.</p>
<p>This is previously assembled genome that was made puplically available. There are several different vendors that provide reference genomes. In general either one of these three will is a good place to start looking for you reference genome.</p>
<ul>
<li><a href="https://asia.ensembl.org/index.html">Ensembl</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a></li>
<li><a href="http://hgdownload.soe.ucsc.edu/downloads.html">UCSC genomes</a></li>
</ul>
<p>Those will have majority of species, however it is common to have a stand along community that host reference genomes from a different place. A good example is <a href="https://www.yeastgenome.org/">yeast genome database</a>. If you are working with yeast this is the place to look for reference genomes.</p>
<p>It is important to note that reference genome isn’t a defined sequence. It is simply the best we have so far and it is a continuos effort to improve quality of the genomes. For widely used model organisms like mouse and yeast their genomes have been sequenced and assembled many times over the years and we have very good quality genomes. Species like axolotl or spiny mouse are less commonly used as model organism in comparison to mouse genomes and quality of those genomes is poorer. There are however other factors in place that influence genome quality e.g genome complexety.</p>
</div>
<div id="exercise" class="section level2">
<h2><span class="header-section-number">3.2</span> Exercise</h2>
<ul>
<li>go to <a href="https://asia.ensembl.org/index.html">Ensembl website</a></li>
<li>from the drop down on the left hand site select <em>Saccharomyces_cerevisiae</em>, which is a yeast species.</li>
<li>click on download FASTA file. This will redirect you to ftp site</li>
<li>find file <strong>Saccharomyces_cerevisiae.R64-1-1.dna_sm.toplevel.fa.gz</strong> and download it by clicking</li>
<li>once downloaded double click on the file to open it and view the content.</li>
</ul>
<p>depending on the operating system that you have you might have to unzip fasta file first before opening. It is best not to open with MS Office suite. Use text editor instead. On windows it’ll be <code>notepad</code> on macOS it’ll <code>textmate</code>.</p>
<p>At the very top you should see this</p>
<pre><code>&gt;I dna_sm:chromosome chromosome:R64-1-1:I:1:230218:1 REF
ccacaccacacccacacacccacacaccacaccacacaccacaccacacccacacacaca
caTCCTAACACTACCCTAACACAGCCCTAATCTAACCCTGGCCAACCTGTCTCTCAACTT
ACCCTCCATTACCCTGCCTCCACTCGTTACCCTGTCCCATTCAACCATACCACTCCGAAC</code></pre>
<p>Note the very first line.</p>
<pre><code>&gt;I dna_sm:chromosome chromosome:R64-1-1:I:1:230218:1 REF</code></pre>
<p>This is a FASTA header that indicates which sequence is follows. In this case it is <code>I</code> i.e sequence from chromosome one of yeast. If there are multiple chromosomes (usually the case for eukaryotic species) then you keep reading the sequence until you hit either the end of the file OR the next “&gt;” sign</p>
<p><strong>NOTE the actual sequnce have been truncated</strong></p>
<pre><code>&gt;I dna_sm:chromosome chromosome:R64-1-1:I:1:230218:1 REF
ccacaccacacccacacacccacacaccacaccacacaccacaccacacccacacacaca
caTCCTAACACTACCCTAACACAGCCCTAATCTAACCCTGGCCAACCTGTCTCTCAACTT
ACCCTCCATTACCCTGCCTCCACTCGTTACCCTGTCCCATTCAACCATACCACTCCGAAC
&gt;II dna_sm:chromosome chromosome:R64-1-1:II:1:813184:1 REF
CCCACACACCACACCCACACCACACCCACACACCACACACACCACACCCACACACCCACA
CCACACCACACCCACACCACACCCACACACCCACACCCACACACCACACCCACACACACC
ACACCCACACACACCCACACCCACACACCACACCCACACACACACCACACCCACACACAC
CACACCACACCCACACCACACCCACACCCACACACCACACCCACACCCACACCCCACACC</code></pre>
<p>The format of the FASTA header is rather loosely defined, but these are key points:</p>
<ul>
<li>must start with “&gt;” sign</li>
<li>the contig/chromsome name follows straight after “&gt;” sign. Can be any string</li>
<li>anything after a space is a comment/description</li>
</ul>
<p>Below are all valid FASTA headers</p>
<pre><code>&gt;I</code></pre>
<pre><code>&gt;chrI</code></pre>
<pre><code>&gt;NC_000067.6</code></pre>
<pre><code>&gt;I_named_it</code></pre>
<p>Each header in a single FASTA file has to be uqniue in order to identify each sequence.</p>
<blockquote>
<p>Are this two FASTA header the same?</p>
</blockquote>
<pre><code>&gt;I</code></pre>
<pre><code>&gt;I named it</code></pre>
<blockquote>
<p>Are this two FASTA header unique?</p>
</blockquote>
<pre><code>&gt;I</code></pre>
<pre><code>&gt;I_named_it</code></pre>
<p>While this apper to be very are small details, the reason it is important because you want to know which chromosome (molecule) your reads came from.</p>
</div>
<div id="rule-of-thumb" class="section level2">
<h2><span class="header-section-number">3.3</span> Rule of thumb</h2>
<p>Unless you are working with some exotic species and/or have a strong prefernce for genome vendor you can generally leave this part up to your bioinformatician to sort out. It is important to know that there are different vendors that have different versions (builds) of genomes.</p>
<p>. Often times bacterial RNAseq data come with some custome reference and annotation files. If that happens bioinformatician will need to convert everythign to FASTA and GTF/GFF files.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bioinformatics-file-formats.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gene-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["rnaseq-intro.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
